velox/serializers/UnsafeRowSerializer.cpp

```diff
@@ -48,11 +48,9 @@ class UnsafeRowVectorSerializer : public VectorSerializer {
       return;
     }
 
-    // Try to allocate 'totalSize' bytes, but handle cases where allocated size
-    // is less.
-    ByteRange byteRange;
-    streamArena_->newRange(totalSize, &byteRange);
-    VELOX_CHECK_EQ(byteRange.position, 0);
+    auto* buffer = (char*)mappedMemory_->allocateBytes(totalSize);
+    buffers_.push_back(
```

Comment:
nit: consider to move buffer insertion to at the end of the function as the buffer has all been used at that point.